### PR TITLE
[circleci] normalize machine types, normalize shell env setup, re-enable code coverage - no move env right now.   not good.

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Make the executor names transparent.    Make the shell setup readable.   Re-enable code coverage.

### Have you read the [Contributing Guidelines on pull requests]

Yes

## Test Plan

CI, and a test branch for coverage, will post back here with it.
Test-1 branch build:   

https://app.circleci.com/pipelines/github/diem/diem/36379/workflows/cf8cf36d-8bf1-448b-bba9-b6ecb2ae1be0/jobs/270030

Let's see if it completes

## Related PRs

Young's pr.

### CODE REVIEW

**Comment by rexhoffman** [comment]
Currently takes about 3 minutes to build install sccache.  Would be great to get x build pulled out the critical path and binaries stored somewhere, same with sccache.   As far as I can tell this isn't supported in any way by cargo -- we'd need a tool like rustup.

**Comment by rexhoffman** [comment]
At this point move tools properly install (21 seconds) and coverage is a couple of hours in to its run.    This should be good, please stamp so I can land in a few hours.

**Comment by rexhoffman** [comment]
Testing coverage back to correct values:

https://app.circleci.com/pipelines/github/diem/diem/36411/workflows/2bc4d1eb-4706-47ee-80a6-9724a4e132c5/jobs/270382

Currently takes 4 hrs 20 minutes, reminder that circleci has a hard cut at 5 hrs.

Notes:  
1) don't try to bump the version of ubuntu we're running on -- this will need more investigation.
2) Set timeout for standard jobs to 70 minutes
3) move lang tooling/testing/coverage is back:

File: .circleci/config.yml [hunk 1]
```diff
@@ -133,7 +112,12 @@ commands:
             echo 'export CARGO='$(rustup which cargo --toolchain "$RUST_NIGHTLY") >> $BASH_ENV
             # Pin the version of RUSTC used for all invocations of cargo
             echo 'export RUSTUP_TOOLCHAIN="$(cat rust-toolchain)"' >> $BASH_ENV
-
+  move_shell_setup:
+    description: Shell Setup
+    steps:
+      - run:
+          name: Setup Env
```
Comment (sausagee) [inline .circleci/config.yml:119]: nit - there is a step in "shell_setup" using the same name.  Consider using a different name for this step, for example "Move Prover setup".

**Review by sausagee — APPROVED** [review]
Looks good!

**Pull request closed without merging**